frontend/src/Authentication.js

```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import { Button, FormGroup, FormControl, ControlLabel, Component } from 'react';
+import axios from 'axios';
+import './Authentication.css'
+
+class Authentication extends Component {
+
+    state = {
+        username:"",
+        password:""
+    };
+
+    validateForm() {
+        return this.state.username.length > 0 && this.state.password.length > 0;
+    }
+
+    async Usernamebtn(){
+        let response =await axios.post('obtain-token/',{username:this.state.username, password:this.state.password})
+    }
```

Comment:
move api calls to separate file. Components should contain only  methods which are strictly related to them and to the DOM. More than one component can use this call so for example in file src/services/authClient.js make function
```
const obtainToken =async (credentials) => {
let response =await axios.post('obtain-token/',{credentials.username, credentials.password})
return response
}
```
and here just use it.

Reply:
also, you need to storage token from response somewhere, localStorage is ok. And remember to add it to axios default headers

Reply:
> also, you need to storage token from response somewhere, localStorage is ok. 

Token is stored in cookie, and attached to http requests automatically.